ArduCopter/mode.cpp

```diff
@@ -516,7 +516,7 @@ void Mode::get_pilot_desired_lean_angles(float &roll_out_cd, float &pitch_out_cd
 // transform pilot's roll or pitch input into a desired velocity
 Vector2f Mode::get_pilot_desired_velocity(float vel_max) const
 {
-    Vector2f vel;
+    Vector2f vel{0.0f, 0.0f};
```

Comment:
These variables are implicitly zero.

https://github.com/ardupilot/ardupilot/blob/master/libraries/AP_Math/vector2.h#L51

Simiarly for Location.

It's just a bit of Magic you have to know about.

Closing - thanks anyway!